controllers/clusterresources/awsendpointserviceprincipal_controller.go

```diff
@@ -0,0 +1,185 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package clusterresources
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/record"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	clusterresourcesv1beta1 "github.com/instaclustr/operator/apis/clusterresources/v1beta1"
+	"github.com/instaclustr/operator/pkg/instaclustr"
+	"github.com/instaclustr/operator/pkg/models"
+	"github.com/instaclustr/operator/pkg/scheduler"
+)
+
+// AWSEndpointServicePrincipalReconciler reconciles a AWSEndpointServicePrincipal object
+type AWSEndpointServicePrincipalReconciler struct {
+	client.Client
+	Scheme        *runtime.Scheme
+	API           instaclustr.API
+	Scheduler     scheduler.Interface
+	EventRecorder record.EventRecorder
+}
+
+//+kubebuilder:rbac:groups=clusterresources.instaclustr.com,resources=awsendpointserviceprincipals,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=clusterresources.instaclustr.com,resources=awsendpointserviceprincipals/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=clusterresources.instaclustr.com,resources=awsendpointserviceprincipals/finalizers,verbs=update
+
```

Comment:
`//+kubebuilder:rbac:groups="",resources=events,verbs=create;patch` is need to be added